gcloud-java-core/src/main/java/com/google/gcloud/Service.java

```diff
@@ -16,6 +16,6 @@
 
 package com.google.gcloud;
 
-public interface Service<O extends ServiceOptions> {
-  O options();
+public interface Service<OptionsT extends ServiceOptions<?, OptionsT>> {
```

Comment:
Google-originated open-source projects use Google style, as Arie linked in the first comment. I found several hundred type variables following this convention in Google's internal copies of these projects.